File: 0645-set-mismatch/0645-set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int n=nums.size();
        unordered_map<int,int> mp;

        for(int i=0;i<n;i++){
            mp[nums[i]]++;
        }
        vector<int> ans;

        for(auto it:mp){
            if(it.second>1) ans.push_back(it.first); //repeating element
        }

        for(int i=1;i<=n;i++){
            if(mp.find(i)==mp.end()) ans.push_back(i); //missing element
        }

        return ans;
    }
};

File: 0645-set-mismatch/0645-set-mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {

        int n=nums.length;
        HashMap<Integer,Integer> mp=new HashMap<>();

        for(int i=0;i<nums.length;i++){
            mp.put(nums[i],mp.getOrDefault(nums[i],0)+1);
        }

        int missing=-1;
        int duplicate=-1;

        for(Map.Entry<Integer,Integer> entry: mp.entrySet()){
            if(entry.getValue()>1) duplicate=entry.getKey();
        }

        for(int i=1;i<=n;i++){
            if(!mp.containsKey(i)) missing=i;
        }

        return new int[]{duplicate,missing};
    }
}